examples/tutorials/nb_python/material_tutorial.py

```diff
@@ -0,0 +1,198 @@
+    agent_cfg.sensor_specifications = [sensor_cfg]
+
+    return habitat_sim.Configuration(backend_cfg, [agent_cfg])
+
+
+def show_obs(sim):
+    data = sim.get_sensor_observations()["rgba_camera"]
+    plt.figure(figsize=(12, 12))
+    plt.imshow(data, interpolation="nearest")
+    plt.axis("off")
+    plt.show(block=False)
+    plt.pause(1)
+
+
+if __name__ == "__main__":
+
+# %%
+    # set up the scene and agent (camera position)
+
+    cfg = make_configuration()
+    sim = habitat_sim.Simulator(cfg)
+    place_agent(sim)
+
+# %%
+    # place a sphere in the scene
+    obj_templates_mgr = sim.get_object_template_manager()
+
+    sphere_template_id = obj_templates_mgr.load_object_configs(
+        str(os.path.join(data_path, "test_assets/objects/sphere"))
+    )[0]
+
+    sphere_ids = []
+    sphere_ids.append(sim.add_object(sphere_template_id))
+    sim.set_translation([3.7, 0.23, 0.0], sphere_ids[0])
+
+    show_obs(sim)
+
+# %%
+    # change the material for the sphere render asset
+
+    # could also use sim.get_object_template(sphere_template_id) instead of
+    # get_object_initialization_template
+    render_asset_handle = sim.get_object_initialization_template(
+        sphere_ids[0]
+    ).render_asset_handle
+    num_materials = sim.get_num_render_asset_materials(render_asset_handle)
+    assert num_materials == 1
+    material_index = 0
+
+    material = sim.get_render_asset_material(render_asset_handle, material_index)
+
+    # make the specular highlights less shiny (more matte)
+    material.shininess *= 0.1
+
+    # reduce the specular highlights (make them darker)
+    material.specular_color *= 0.3
+
+    # set diffuse to dark green
+    material.diffuse_color = mn.Color4(r=0.2, g=0.4, b=0.2)
+
+    # change material for all existing and future uses of this render asset, including
+    # the existing sphere.
+    sim.set_render_asset_material(render_asset_handle, material_index, material)
+
+    show_obs(sim)
+
+# %%
+    # create more spheres. They will also use the modified material.
+
+    for x in [3.0, 2.3, 1.6, 0.9]:
+        id = sim.add_object(sphere_template_id)
+        sim.set_translation([x, 0.23, 0.0], id)
+        sphere_ids.append(id)
+
+    show_obs(sim)
+
+# %%
+    # override_object_render_asset_material API is still pending. I leave this
+    # code for reference.
+    if False:
+        # randomize material properties of individual spheres
+        random.seed(5)
+        for id in sphere_ids:
+            randomized_material = PhongMaterialInfo(material)
+            randomized_material.specular_color *= random.uniform(0, 2)
+            randomized_material.diffuse_color = mn.Color4(
+                r=material.diffuse_color.r + random.uniform(-0.1, 0.1),
+                g=material.diffuse_color.g + random.uniform(-0.1, 0.1),
+                b=material.diffuse_color.b + random.uniform(-0.1, 0.1),
+            )
+            sim.override_object_render_asset_material(
```

Comment:
This is probably the strangest API here and most likely to change, so I just removed it from this PR. See this slack discussion: https://cvmlp.slack.com/archives/G0131KVLBLL/p1597770680011800 .

This PR still has the C++ backend, `Simulator::overrideObjectRenderAssetMaterial`, which is unfortunately not tested at all in this PR.